src/main/java/duke/command/DeleteCommand.java

```diff
@@ -0,0 +1,68 @@
+package duke.command;
+
+import duke.command.Command;
+import duke.main.Ui;
+import duke.utils.Parser;
+import duke.utils.Storage;
+import duke.utils.TaskList;
+import duke.utils.Utils;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class DeleteCommand implements Command {
+
+    @Override
+    public void execute(String task, Ui ui, Storage storage, TaskList taskList) {
+        String[] token = task.split(" ", 2);
+        if (token.length < 2) {
+            ui.setResponse("Please specify which task to delete");
+            return;
+        }
+        String[] indices = token[1].split(" ");
+        if (!Utils.isNumeric(indices[0])) {
+            ui.setResponse("Please specify which task to delete");
+            return;
+        }
+        List<Integer> toBeDeleted = new ArrayList<>();
+        for (int i = 0; i < indices.length; i++) {
+            if (Utils.isNumeric(indices[i]) != true) {
+                continue;
+            } else {
+                toBeDeleted.add(Integer.parseInt(indices[i]) - 1);
+            }
+        }
+        Collections.sort(toBeDeleted);
+        if (toBeDeleted.size() < 1) {
+            ui.setResponse("Please specify which task to delete");
+            return;
+        }
+        if (toBeDeleted.get(0) >= taskList.size() || toBeDeleted.get(toBeDeleted.size() - 1) < 0) {
+            ui.setResponse("No such task number");
+            return;
+        }
+        String response = "";
```

Comment:
May be better to leave a line here, as with other parts of the code you deem fit - this helps in readability as it separates the logical parts of your code based on their purpose.